modules/principals/src/main/frontend/src/Userboard/Groups/addusertogroupdialogue.jsx

```diff
@@ -0,0 +1,187 @@
+
+import MaterialTable from 'material-table';
+
+class AddUserToGroupDialogue extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            allUserNames: [],
+            freeUserNames: [],
+            selectedUsers: []
+        }
+    }
+
+    addName(name) {
+        return { name }
+    }
+
+    handleLoadUsers() {
+        fetch("/system/userManager/user.1.json",
+            {
+                method: 'GET',
+                headers: {
+                    'Authorization': 'Basic ' + btoa('admin:admin')
+                }
+            })
+            .then((response) => {
+                return response.json();
+            })
+            .then((data) => {
+                var names = [];
+                for (var username in data) {
+                    !username.startsWith("sling-") && names.push(this.addName(username));
+                }
+                this.setState({ allUserNames: names });
+            })
+            .catch((error) => {
+                console.log(error);
+            });
+    }
+
+    handleAddUsers() {
+        let url = "/system/userManager/group/" + this.props.name + ".update.html";
+        let formData = new FormData();
+
+        var i;
+        for (i = 0; i < this.state.selectedUsers.length; ++i) {
+            formData.append(':member', this.state.selectedUsers[i]);
+        }
+
+        fetch(url,
+            {
+                method: 'POST',
+                credentials: 'include',
+                body: formData
+            })
+            .then(() => {
+                this.setState({ selectedUsers: [] });
+                this.props.reload();
+                this.handleExit();
+            })
+            .catch((error) => {
+                console.log(error);
+            });
+    }
+
+    componentDidMount() {
+        this.handleLoadUsers();
+    }
+
+    // TODO - find more efficient way to add and remove users from list
+    handleSelectRowClick(rows) {
+        let chosens = [];
+        var i;
+        for (i = 0; i < rows.length; ++i) {
+            chosens.push(rows[i].name);
+        }
+        this.setState({ selectedUsers: chosens });
+    }
+
+    handleEntering() {
+        if (this.props.name != "") {
+          fetch("/system/userManager/group/" + this.props.name + ".1.json",
+            {
+              method: 'GET',
+              credentials: 'include'
+            })
+            .then((response) => {
+              return response.json();
+            })
+            .then((data) => {
+              var groupUsers = [];
+              var i;
+              for (i = 0; i < data.members.length; ++i) {
+                let username = data.members[i];
+                username = username.split('/').pop();
+                groupUsers.push(username);
+              }
+
+              let filtered = [];
+              filtered = this.state.allUserNames.filter( function( el ) {
```

Comment:
We should be consistent in our usage of arrow functions.
```suggestion
              filtered = this.state.allUserNames.filter( (el) => {
```